src/Controls/src/Core/Application/Application.Windows.cs

```diff
@@ -1,6 +1,147 @@
-﻿namespace Microsoft.Maui.Controls
+﻿using System;
+using Microsoft.Maui.ApplicationModel;
+using Microsoft.UI.Windowing;
+using Microsoft.UI.Xaml;
+using Windows.UI;
+
+namespace Microsoft.Maui.Controls
 {
 	public partial class Application
 	{
+		AppTheme? _currentThemeForWindows;
+
+		partial void OnRequestedThemeChangedPlatform(AppTheme newTheme)
+		{
+			_currentThemeForWindows = newTheme;
+
+			ApplyThemeToAllWindows(newTheme, UserAppTheme == AppTheme.Unspecified);
+		}
+
+		partial void OnPlatformWindowAdded(Window window)
+		{
+			window.HandlerChanged += OnWindowHandlerChanged;
+
+			if (_currentThemeForWindows is not null && window.Handler is not null)
+			{
+				TryApplyThemeToWindow(window);
+			}
+		}
+
+		partial void OnPlatformWindowRemoved(Window window)
+		{
+			window.HandlerChanged -= OnWindowHandlerChanged;
+		}
+
+		void OnWindowHandlerChanged(object? sender, EventArgs e)
+		{
+			if (sender is Window window)
+			{
+				TryApplyThemeToWindow(window);
+			}
+		}
+
+		void TryApplyThemeToWindow(Window window)
+		{
+			if (_currentThemeForWindows is AppTheme theme)
+			{
+				var forcedElementTheme = GetElementTheme();
+
+				if (IsWindowReady(window))
+				{
+					ApplyThemeToWindow(window, UserAppTheme == AppTheme.Unspecified, forcedElementTheme);
+				}
+			}
+		}
+
+		bool IsWindowReady(Window window)
+		{
+			var platformWindow = window?.Handler?.PlatformView as UI.Xaml.Window;
+			return platformWindow?.Content is FrameworkElement;
+		}
+
+		void ApplyThemeToAllWindows(AppTheme newTheme, bool followSystem)
+		{
+			var forcedElementTheme = GetElementTheme();
+
+			foreach (var window in Windows)
+			{
+				if (IsWindowReady(window))
+				{
+					ApplyThemeToWindow(window, followSystem, forcedElementTheme);
+				}
+			}
+		}
+
+		ElementTheme GetElementTheme()
+		{
+			if (_currentThemeForWindows is AppTheme theme)
+			{
+				return theme switch
+				{
+					AppTheme.Dark => ElementTheme.Dark,
+					AppTheme.Light => ElementTheme.Light,
+					_ => ElementTheme.Default
+				};
+			}
+			return ElementTheme.Default;
+		}
+
+		void ApplyThemeToWindow(Window? window, bool followSystem, ElementTheme forcedElementTheme)
+		{
+			var platformWindow = window?.Handler?.PlatformView as UI.Xaml.Window;
+
+			if (platformWindow is null)
+			{
+				System.Diagnostics.Debug.WriteLine("ApplyThemeToWindow: platformWindow is null. Unable to apply theme to the root element.");
+				return;
+			}
+
+			platformWindow.DispatcherQueue.TryEnqueue(() =>
```

Comment:
Could also check if `(platformWindow.DispatcherQueue is null)`? Not a common case, but could happen if the Window is disposed.

Reply:
@jsuarezruiz, okay, as suggested, I have added a null check for `platformWindow.DispatcherQueue` and return early if it’s null.